python/tank/deploy/descriptor.py

```diff
@@ -450,30 +453,55 @@ def get_from_location(app_or_engine, pipeline_config, location_dict):
     from .git_descriptor import TankGitDescriptor
     from .manual_descriptor import TankManualDescriptor
 
+    if not isinstance(pipeline_config, dict):
```

Comment:
Are you trying to stay backwards compatible here?

I would suggest a slightly different approach, having looked at the code. It seems the methods `pipeline_config.get_apps_location()`, `pipeline_config.get_engine_location()` etc are only used in one place. I suggest we change the method signature to:

```
 def get_from_location(app_or_engine, pipeline_config_path, bundle_install_path, location_dict):
     """
     Factory method.
 
     :param app_or_engine: Either AppDescriptor.APP AppDescriptor.ENGINE or FRAMEWORK
     :param pipeline_config_path: Path to the current pipeline configuration 
     :param install_path: Root path to bundle install location
     :param location_dict: A tank location dict
     :returns: an AppDescriptor object
     """
```

Get rid of the `pipeline_config.get_xxx_location()` and take it from there. Does that make sense? It feels a lot clearer to me.